src/actions/tree.ts

```diff
@@ -61,16 +71,23 @@ export const applyInViewNodesToTree = (idx, tree) => {
  * this fn relies on the "inView" attr of nodes
  * note that this function checks to see if the tree has been defined (different to if it's ready / loaded!)
  * for arg destructuring see https://simonsmith.io/destructuring-objects-as-function-parameters-in-es6/
- * @param  {array|undefined} root Change the in-view part of the tree. [root idx tree1, root idx tree2].
- *                                [0, 0]: reset. [undefined, undefined]: do nothing
- * @param  {object | undefined} tipSelected
- * @param  {string | undefined} cladeSelected
- * @return {function} a function to be handled by redux (thunk)
+ * Returns a function to be handled by redux (thunk)
  */
-export const updateVisibleTipsAndBranchThicknesses = (
-  {root = [undefined, undefined], cladeSelected = undefined} = {}
-) => {
-  return (dispatch, getState) => {
+export const updateVisibleTipsAndBranchThicknesses = ({
+  root = [undefined, undefined],
+  cladeSelected = undefined,
+}: {
+  /**
+   * Change the in-view part of the tree.
+   *
+   * [0, 0]: reset. [undefined, undefined]: do nothing
+   */
+  root?: Root
+
+  cladeSelected?: string
+} = {}
+):       (dispatch: AppDispatch, getState: () => RootState) => void => {
+  return (dispatch: AppDispatch, getState: () => RootState) => {
```

Comment:
As commented elsewhere, these duplicate function types would be nice to avoid, unless you think they are somehow advantageous.